Doctor flags missing_tasks_dir as ERROR for unplanned future slices (regression from #2017/#2323)

## Problem

After `plan-milestone`, doctor flags `missing_slice_dir` and `missing_tasks_dir` as severity **error** for every roadmap slice that hasn't been planned yet. These directories are created lazily by `ensurePreconditions()` at dispatch time — their absence before `plan-slice` runs is expected, not broken. The false errors degrade health trend scores, trigger "Some issues detected" warnings in the dashboard, and increment the `consecutiveErrorUnits` counter toward unnecessary LLM-heal escalation.

This is the same bug reported in #2017 (v2.41.0) and #2323 (v2.43.0). PR #2042 was the proposed fix but was **closed without merging**. PR #2280 was claimed to resolve it by moving to DB-backed slice discovery, but `plan-milestone` inserts all slice rows with `status: "pending"`, so `getMilestoneSlices()` still returns them all.

## Root Cause

**`doctor.js:~551-565`** — The `missing_tasks_dir` check uses a binary severity branch:

```javascript
const tasksDir = resolveTasksDir(basePath, milestoneId, slice.id);
if (!tasksDir) {
    issues.push({
        severity: slice.done ? "warning" : "error",  // ← no "pending/not started" state
        code: "missing_tasks_dir",
        ...
    });
}
```

Same defect at **`doctor.js:~531-549`** for `missing_slice_dir`.

The DB mapping at **`doctor.js:~443`** collapses status into boolean `done`, losing the `pending` vs `active` distinction:

```javascript
slices = dbSlices.map(s => ({
    id: s.id,
    title: s.title,
    done: s.status === "complete",  // "pending" maps to done:false, same as "active"
    risk: (s.risk || "medium"),
    depends: s.depends,
    demo: s.demo,
}));
```

The interaction chain:
1. `plan-milestone` inserts all slices into DB with `status: "pending"` (`tools/plan-milestone.js:~167`)
2. Post-unit doctor runs on milestone scope (`auto-post-unit.js:~133`)
3. Doctor's auto-fix creates empty slice dirs for `missing_slice_dir`
4. Next doctor pass finds slice dirs without `tasks/` subdirs → `missing_tasks_dir` ERROR
5. `doctor-proactive.js:66` increments `consecutiveErrorUnits`
6. Dashboard shows yellow health ("Some issues detected") with false error count

## Expected Behavior

A missing directory for a not-yet-planned slice should be **skipped** or severity **info**, not **error**. The `tasks/` directory is created on-demand by `ensurePreconditions()` when the slice is dispatched.

### Suggested Fix

Gate directory checks behind a materialization condition — a slice is "materialized" if it has a plan file or is already complete:

```javascript
// Before the slicePath/tasksDir checks:
const hasPlan = !!resolveSliceFile(basePath, milestoneId, slice.id, "PLAN");
const isMaterialized = slice.done || hasPlan;

// Then:
if (!slicePath) {
    if (!isMaterialized) continue;  // Future slice — absence is expected
    issues.push({ severity: slice.done ? "warning" : "error", ... });
}

// Same for tasksDir:
if (!tasksDir) {
    if (!isMaterialized) continue;
    issues.push({ severity: slice.done ? "warning" : "error", ... });
}
```

Alternatively, preserve the DB status in the mapping:

```javascript
slices = dbSlices.map(s => ({
    ...
    done: s.status === "complete",
    pending: s.status === "pending",  // new field
}));
// Then: if (slice.pending && !slicePath) continue;
```

## Environment

- GSD version: 2.44.0
- Model: claude-opus-4-6
- Unit: plan-milestone/M002

## Reproduction Context

Phase: milestone planning for M002. `plan-milestone M002` wrote ROADMAP with slices S01-S03. Post-unit doctor ran on scope M002, flagged all slice directories as missing, auto-created empty dirs, then on next pass flagged `missing_tasks_dir` for S02 and S03 (S01 was materialized by `plan-slice`).

## Forensic Evidence

Doctor issues from v2.44.0 session:
```
[ERROR] M002/S02 | missing_tasks_dir | Missing tasks directory for M002/S02
[ERROR] M002/S03 | missing_tasks_dir | Missing tasks directory for M002/S03
```

Dashboard showed: "1 consecutive error unit(s) · Missing slice directory for M002/S01 · Missing slice directory for M002/S02"

Disk state confirms S02/S03 are empty dirs (no plan, no tasks — expected for unplanned slices):
```
slices/S02/: (empty directory)
slices/S03/: (empty directory)
```

## Prior Issues

- #2017 — same bug, closed as "obsolete" (v2.41.0)
- #2323 — same bug, closed as "fixed by #2042" (v2.43.0)
- PR #2042 — the actual fix, **closed without merging**
- PR #2280 — claimed to fix via DB migration, did not address the root cause

---
*Auto-generated by `/gsd forensics`*

--- a/src/resources/extensions/gsd/doctor.ts
+++ b/src/resources/extensions/gsd/doctor.ts
@@ -470,7 +470,7 @@ export async function runGSDDoctor(basePath: string, options?: { fix?: boolean; 
     if (!roadmapContent) continue;
 
     // Normalize slices: prefer DB, fall back to parser
-    type NormSlice = RoadmapSliceEntry;
+    type NormSlice = RoadmapSliceEntry & { pending?: boolean };
     let slices: NormSlice[];
     if (isDbAvailable()) {
       const dbSlices = getMilestoneSlices(milestoneId);
@@ -478,6 +478,7 @@ export async function runGSDDoctor(basePath: string, options?: { fix?: boolean; 
         id: s.id,
         title: s.title,
         done: s.status === "complete",
+        pending: s.status === "pending",
         risk: (s.risk || "medium") as RoadmapSliceEntry["risk"],
         depends: s.depends,
         demo: s.demo,
@@ -564,6 +565,9 @@ export async function runGSDDoctor(basePath: string, options?: { fix?: boolean; 
 
       const slicePath = resolveSlicePath(basePath, milestoneId, slice.id);
       if (!slicePath) {
+        // Pending slices haven't been planned yet — directories are created
+        // lazily by ensurePreconditions() at dispatch time. Skip them.
+        if (slice.pending) continue;
         const expectedPath = relSlicePath(basePath, milestoneId, slice.id);
         issues.push({
           severity: slice.done ? "warning" : "error",
@@ -586,6 +590,8 @@ export async function runGSDDoctor(basePath: string, options?: { fix?: boolean; 
 
       const tasksDir = resolveTasksDir(basePath, milestoneId, slice.id);
       if (!tasksDir) {
+        // Pending slices haven't been planned yet — tasks/ is created on demand.
+        if (slice.pending) continue;
         issues.push({
           severity: slice.done ? "warning" : "error",
           code: "missing_tasks_dir",

